polygeist/tools/cgeist/Lib/ValueCategory.cc

```diff
@@ -290,3 +290,59 @@ ValueCategory ValueCategory::FPExt(OpBuilder &Builder,
   warnUnconstrainedCast<arith::ExtFOp>();
   return Cast<arith::ExtFOp>(Builder, PromotionType);
 }
+
+ValueCategory ValueCategory::SIToFP(OpBuilder &Builder,
+                                    Type PromotionType) const {
+  warnUnconstrainedCast<arith::SIToFPOp>();
+  return Cast<arith::SIToFPOp>(Builder, PromotionType);
+}
+
+ValueCategory ValueCategory::UIToFP(OpBuilder &Builder,
+                                    Type PromotionType) const {
+  warnUnconstrainedCast<arith::UIToFPOp>();
+  return Cast<arith::UIToFPOp>(Builder, PromotionType);
+}
+
+ValueCategory ValueCategory::FPToUI(OpBuilder &Builder,
+                                    Type PromotionType) const {
+  warnUnconstrainedCast<arith::FPToUIOp>();
+  return Cast<arith::FPToUIOp>(Builder, PromotionType);
+}
+
+ValueCategory ValueCategory::FPToSI(OpBuilder &Builder,
+                                    Type PromotionType) const {
+  warnUnconstrainedCast<arith::FPToSIOp>();
+  return Cast<arith::FPToSIOp>(Builder, PromotionType);
+}
+
+ValueCategory ValueCategory::IntegerCast(OpBuilder &Builder, Type PromotionType,
+                                         bool IsSigned) const {
+  auto SrcIntTy = val.getType().dyn_cast<IntegerType>();
+  auto DstIntTy = PromotionType.dyn_cast<IntegerType>();
```

Comment:
dyn_cast will return NULL if the operand cannot be casted. Lines 323, 324 use SecIntTy and DstIntTy and may dereference a NULL pointer.

Reply:
True, that was supposed to be a plain `cast`.